heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/AppsV1Controller.java

```diff
@@ -97,20 +95,13 @@ boolean submit(PackingPlan packingPlan) {
     for (PackingPlan.ContainerPlan containerPlan : packingPlan.getContainers()) {
       numberOfInstances = Math.max(numberOfInstances, containerPlan.getInstances().size());
     }
-    final V1beta1StatefulSet statefulSet = createStatefulSet(containerResource, numberOfInstances);
+    final V1StatefulSet statefulSet = createStatefulSet(containerResource, numberOfInstances);
 
     try {
-      final Response response =
-          client.createNamespacedStatefulSetCall(getNamespace(), statefulSet, null,
-              null, null).execute();
-      if (!response.isSuccessful()) {
-        LOG.log(Level.SEVERE, "Error creating topology message: " + response.message());
-        KubernetesUtils.logResponseBodyIfPresent(LOG, response);
-        // construct a message based on the k8s API server response
-        throw new TopologySubmissionException(
-            KubernetesUtils.errorMessageFromResponse(response));
-      }
-    } catch (IOException | ApiException e) {
+      final V1StatefulSet response =
+          client.createNamespacedStatefulSet(getNamespace(), statefulSet, null,
```

Comment:
Should we check if the response is successful here?

Reply:
I believe the `ApiException` will have the status code because the `ApiClient` class checks the status code and throws the exception if there is an error. Is there any better logging you would suggest in the catch block when we deal with the `ApiException`?

Here is a link to the [`ApiClient.handleResponse()`](https://github.com/kubernetes-client/java/blob/d94b271a7dc9636d1b5d930b047b94273b58aa83/kubernetes/src/main/java/io/kubernetes/client/openapi/ApiClient.java#L954) logic.

Reply:
Nothing off the top of my head that would make it any better.  It looks like the client handles printing out enough information.  We can always come back and add more logging if we run into issues or someone requests it.